app/src/js/server/create_project.ts

```diff
@@ -79,35 +79,72 @@ export async function parseForm (
  * Parses item, category, and attribute files from paths
  */
 export async function parseFiles (
-  labelType: string, files: { [key: string]: string }, itemsRequired: boolean)
+  labelType: string, files: { [key: string]: string },
+  storage: Storage, itemsRequired: boolean)
   : Promise<FormFileData> {
-  return Promise.all([
-    parseItems(files, itemsRequired),
-    parseSensors(files),
-    parseTemplates(files),
-    parseAttributes(files, labelType),
-    parseCategories(files, labelType)])
+  const items = parseItems(files, itemsRequired, storage)
+
+  const categories: Promise<Category[]> = readFileWithDefault(
+    files, FormField.CATEGORIES, getDefaultCategories(labelType), storage)
+
+  const sensors: Promise<SensorType[]> = readFileWithDefault(
+    files, FormField.SENSORS, [], storage)
+
+  const templates: Promise<Label2DTemplateType[]> = readFileWithDefault(
+    files, FormField.LABEL_SPEC, [], storage)
+
+  const attributes = readFileWithDefault(files, FormField.ATTRIBUTES,
+    getDefaultAttributes(labelType), storage)
+
+  return Promise.all([items, sensors, templates, attributes, categories])
     .then((result: [
       Array<Partial<ItemExport>>,
       SensorType[],
       Label2DTemplateType[],
       Attribute[],
-      string[]
+      Category[]
     ]) => {
+      const categoriesData = result[4]
+      const categoriesList = []
+      for (const category of categoriesData) {
+        categoriesList.push(category.name)
+      }
       return {
         items: result[0],
         sensors: result[1],
         templates: result[2],
         attributes: result[3],
-        categories: result[4]
+        categories: categoriesList
       }
     })
 }
 
+/** Read and parse a yaml or json file of arbitrary type. */
+export async function readFile<T> (path: string, storage: Storage): Promise<T> {
```

Comment:
It is more intuitive to put `storage` as the first argument. For example, in C file read api, such as `fscanf`, file and storage pointers are usually the first argument. It is the same for the other file reading api.

Reply:
Fixed